### DISCUSSION TITLE
Formula

### DISCUSSION THREAD

**Discussion by DannyHatcher-Git** [RafaelGB/obsidian-db-folder#385] [Q&A] [answered]
The new formula property looks exciting but I don't understand how I can use it. 
I have read through the help documentation but as someone that doesn't code it doesn't help me. Is there an example somewhere?

**Comment by RafaelGB** [comment]
For example, the dates are a Luxon Datetime property, so you can use their properties with this example:
```js
${row.deadline.diffNow("days").toFormat("d")} days left
```
https://user-images.githubusercontent.com/11924043/190251142-6cd9a440-4e50-4185-8f2e-9a19a4da7c35.mov

**Reply by DannyHatcher-Git** [reply]
I have tried putting your code in the formula settings box, in a js file in the folder and changed all the settings but still can't get it to work.

Can you write out step by step what to do I am new to programming and this doesn't make sense to me yet.

**Reply by RafaelGB** [reply]
Could you share a mock sample?

Also you can consult our formulas documentation https://rafaelgb.github.io/obsidian-db-folder/features/Formulas/


Formulas is using the js expressions standard https://developer.mozilla.org/en-US/docs/Learn/JavaScript/First_steps/Strings#including_expressions_in_strings

You can find the available variables for those expressions inside the docu. For our example we are using `row`. Its an Object containing all the values of the cells for that row.

Try first your showing the content of your cells. For example `${row.column1}` will show the content of the column1 cell for all formula rows

**Reply by DannyHatcher-Git** [reply]
I don't find the formula documentation helpful.
It is very similar to the other plugin documentation, written for people that understand programming.
As a beginner, the words written mean nothing to me.

I understand how to write a document with expressions in but I can't seem to get the file to render in the database view.

My approach was to use your example but there was no way that I could get it to work. I have tried typing `${row.column1} into the formula box but nothing happens. 
![image](https://user-images.githubusercontent.com/78496825/194769161-575719c9-d264-42e6-9b82-7160d4733033.png)

I have tried renaming column1 to File, but still nothing.
Yes, I have ticked the enable JavaScript formulas.
The select the formula JS folder is the root as I don't have any JS files at the moment because I thought I could just type it in the box.

**Comment by RafaelGB** [comment, marked as answer]
For our example we don't need to enable JS formulas. Its a direct expression.

Try to run this folder sample
[formulaExample.zip](https://github.com/RafaelGB/obsidian-db-folder/files/9742130/formulaExample.zip)


https://user-images.githubusercontent.com/11924043/194769693-d1298e33-cb19-43d1-b588-525e33f901a1.mov

**Reply by DannyHatcher-Git** [reply]
So I got yours working and did some troubleshooting.
I was using ${row.File} which didn't work I am guessing because it is the main reference column.
I then tried test 1 as the column name and it still didn't work, but when I got rid of the 1 it did. So I assume the row has to be continuous in its reference.

**Reply by RafaelGB** [reply]
Currently `File` column is not avaliable on formulas. Will be supported with rollups in future versions.

About the naming its a little Bir confusing (keeping working of improve the UX). Formulas is using the id of the column, not the label. You can see your column ids inside the markdown ddbb file

**Reply by DannyHatcher-Git** [reply]
Ok thanks!

It says you are a maintainer but I though you were/are the original creator?

**Reply by RafaelGB** [reply]
I am the creator =)
